api/endpoints.go

```diff
@@ -0,0 +1,9 @@
+package api
+
+const (
+	UserEndpoint          = "/update_or_create_user"
+	ReportEndpoint        = "/report"
+	GetMapEndpoint        = "/get_map"
+	ReadReferralEndpoint  = "/readreferral"
+	WriteReferralEndpoint = "/writereferral"
+)
```

Comment:
If you indeed want a separate file and package for "api"s, then structures UserArgs, ReporArgs, etc. also belong here.

If you did so because of client.go / client/main.go than
1. Why do so if this was not necessary fro backend/main.go?
2. clint is esentially a test (just different level test), for tests such separation is normally not done in go.

Reply:
> If you indeed want a separate file and package for "api"s, then structures UserArgs, ReporArgs, etc. also belong here.

It's a good idea, I'm moving all API related structures into the API package. And yes, the client uses only API and never needs anything else from the backend.